src/components/ChannelList/hooks/usePaginatedChannels.ts

```diff
@@ -72,10 +83,34 @@ export const usePaginatedChannels = <
     setQueryInProgress(null);
   };
 
+  const throttleRecover = useCallback(() => {
+    const now = Date.now();
+    const isFirstRecovery = !lastRecoveryTimestamp.current;
+    const timeElapsedSinceLastRecoveryMs = lastRecoveryTimestamp.current
+      ? now - lastRecoveryTimestamp.current
+      : 0;
+
+    if (!isFirstRecovery && timeElapsedSinceLastRecoveryMs < recoveryThrottleInterval && !error) {
+      return;
+    }
+
+    lastRecoveryTimestamp.current = now;
+    queryChannels('reload');
+  }, [error, queryChannels, lastRecoveryTimestamp]);
```

Comment:
```suggestion
  }, [error, queryChannels, recoveryThrottleInterval]);
```

- Refs shall not be put in dependencies
- `recoveryThrottleInterval` isn't listed here.